File: src/test/java/com/erste/util/ResourceFileReaderTest.java
package com.erste.util;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;

public class ResourceFileReaderTest {

    private final ResourceFileReader resourceFileReader = new ResourceFileReader("src/test/resources");

    @Test
    void testReadStopWordsFile() {
        String filename = "stopwords.txt";
        List<String> expectedResult = List.of("a", "the", "off", "on");

        List<String> result = resourceFileReader.readFile(filename);

        Assertions.assertTrue(expectedResult.containsAll(result));
        Assertions.assertEquals(expectedResult.size(), result.size());
    }
}

File: src/test/kotlin/at/erste/ResourceFileReaderTest.kt
package at.erste

import at.erste.io.ResourceFileReader
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import kotlin.test.assertContentEquals

class ResourceFileReaderTest {
    private val resourceFileReader = ResourceFileReader()
    private val skippedWords = listOf("and", "or", "true", "false", "1", "155")

    @Test
    fun testReadNonExistingFile() {
        val actual = resourceFileReader.readByLine("/non-existing.txt")
        assertTrue(actual.isEmpty())
    }

    @Test
    fun testEmptyFile() {
        // empty file is being read as a single line
        val actual = resourceFileReader.readByLine("/empty.txt")
        assertEquals(1, actual.size)
    }

    @Test
    fun testReadFile() {
        val actual = resourceFileReader.readByLine("/valid.txt")
        assertTrue(actual.isNotEmpty())
        assertContentEquals(skippedWords, actual)
    }
}
